src/main/java/org/fiware/iam/tir/auth/VCRolesFinder.java

```diff
@@ -0,0 +1,56 @@
+package org.fiware.iam.tir.auth;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.security.token.DefaultRolesFinder;
+import io.micronaut.security.token.config.TokenConfiguration;
+import jakarta.inject.Singleton;
+import org.fiware.iam.common.model.Role;
+import org.fiware.iam.common.model.VCData;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Singleton
+@Replaces(DefaultRolesFinder.class)
+public class VCRolesFinder extends DefaultRolesFinder {
+    private static final String VC_KEY = "vc";
+
+    @Value("${allowedVCTypes}")
```

Comment:
Why do we need "allowedTypes"? The allowed types should already have been checked by the verifier. Btw. please dont create configuration via @value, add them to a configuration-properties class like GeneralProperties.java